docs/developersetup.md

```diff
@@ -0,0 +1,43 @@
+# Developer Setup
+---
+This page contains information about how to set up your local environment to develop for this project. 
+
+## Overview of Stack
+---
+Homebrew: This is a package manager that lets us install other packages.
+Docker: This will load all the dependencies you need in an “image” so you don’t mess up your local computer. We launch a container and do all our hacking in there.
+Visual Studio Code: This is an IDE that has git/terminal/docker integration. You’ll write your code here
+
+## macOS Setup
+---
+
+1. Install Homebrew with: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+2. Install Docker with: brew install docker
+3. Install Git with: brew install git
+4. Install IDE
+
+## Git 
+---
+
+### Setting up your local repo
+
+1. Clone repo locally with: git clone https://github.com/mellychat/webapp.git 
+2. Install recommended extensions
+3. Setup remote to track upstream with: git remote add upstream https://github.com/mellychat/webapp.git
+
+### Workflow
+
+1. Fetch any changes from the upstream/mainline branch and merges them into your local copy of mainline:
+
+  git fetch upstream
+  git checkout mainline
+  git merge upstream/mainline
+
+2. Create a new branch on which you will make whatever changes you want with: git checkout -b my-feature-branch
+3. Commit any changes you have made to your local branch. Push your changes to your Github with: git push upstream
+4. Send a pull request so that we get a copy of the feature branch you wish to merge 
+5. We merge your feature branch into mainline
+
+
```

Comment:
Our gitflow has changed a bit. So its push upstream <branch name>. 

Also @j0n3lson should we add special commands:

Such as: "git add." or "--force" and etc?

Reply:
Re: Pushing changes:
+1 to what Steve said. It should be `git push upstream my-feature-branch` in this case.

Re: Documenting Git commands: 
1. I don't want to be too prescriptive since working with git is a prereq for anyone using github.
2. We can probably link to a few of the git-related videos I been sending out there. I think a video is worth a thousand words in this case.
3. We should document basic/well known operations (e.g. `git add .`) or commands that only get used sometimes (e.g `--force`). If you don't know the basic commands its probably worth it to stop and watch some videos before proceeding. Likewise, the rarely used commands require a lot more background info before it's safe to use so we don't want to advertise using those commands in our docs.